### PR TITLE
HTML formatting on communication consoles, HTML sanitization

### LINKED ISSUE #54713: Messages shown in communications consoles do not display HTML properly.
For example, messages like what threat level the station is use <b>bold tags</b>, but they instead display the tags directly.

### PR DESCRIPTION
**Pull request by prodirus**
## About The Pull Request

Fixes #54713 

Lets the comms console display HTML by telling react to set the HTML directly.

Also lets admins send HTML formatted command updates to the console. However, if the message is also sent to the crew, the raw HTML string will be sent in the chat box!

Shoutout to Timberpoes for helping me out with this!

Now with sanitize spun off into its own thing for anyone to reuse!

Accepts a string with HTML tags in it as input, and optionally allowed tags/forbidden attributes to override the defaults, then spits out a sanitised version of the same string!

## Why It's Good For The Game

image

Looks better and more readable than text buried in HTML tags.

## Changelog
:cl:
fix: Roundstart Centcomm reports on communication consoles should no longer be word soup!
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/CommunicationsConsole.js [hunk 1]
```diff
@@ -14,6 +15,24 @@ const SWIPE_NEEDED = "SWIPE_NEEDED";
 
 const sortByCreditCost = sortBy(shuttle => shuttle.creditCost);
 
+// Sanitization code stolen from PaperSheet.js, typos and all.
+const sanatize_text = value => {
```
Comment (Jared-Fogle) [inline tgui/packages/tgui/interfaces/CommunicationsConsole.js:19]:
```suggestion
const sanitize_text = value => {
```

File: tgui/packages/tgui/interfaces/CommunicationsConsole.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import { sortBy } from "common/collections";
+import DOMPurify from 'dompurify';
 import { capitalize } from "common/string";
```
Comment (Jared-Fogle) [inline tgui/packages/tgui/interfaces/CommunicationsConsole.js:3]:
```suggestion
import { capitalize } from "common/string";
import DOMPurify from 'dompurify';
```

please i made my imports beautifully sorted in alphabetical order :(

Comment (prodirus) [reply tgui/packages/tgui/interfaces/CommunicationsConsole.js:3]: sorry i will fix :(

**Review by stylemistake — DISMISSED** [review]
Fine solution, although I would pick out `sanitizeText` into its own module for reusability (e.g. `tgui/packages/tgui/sanitize.js`), and yeah, camel case naming is preferable.

**Comment by prodirus** [comment]
> Fine solution, although I would pick out `sanitizeText` into its own module for reusability (e.g. `tgui/packages/tgui/sanitize.js`), and yeah, camel case naming is preferable.

Sounds good, will work on it!

**Comment by prodirus** [comment]
@stylemistake Sanitize spun off into its own module, with optional parameters for forbidden attributed/allowed tags. Also, camelcase all around!

**Review by Cyberboss — APPROVED** [review]
Thanks for the fix

**Pull request merged**